File: javascript/acronym/acronym.js
//
// This is only a SKELETON file for the 'Acronym' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const parse = (string) => {
  return string.split(/[^a-zA-Z']/)
    .map(phrase => phrase[0] && phrase[0].toUpperCase())
    .join('')
};

File: java/acronym/src/main/java/Acronym.java
class Acronym {
    String phrase;
    
    Acronym(String phrase) {
        this.phrase = phrase;
    }

    String get() {
       String[] words = this.phrase.split("\\s|-|_");
       StringBuffer ac = new StringBuffer();
       for(int i=0; i < words.length; i++) {
           if(words[i].length() == 0) continue;
           Character c = words[i].charAt(0);
           if(Character.isLetter(c)) ac.append(c);
       }
       return ac.toString().toUpperCase();
    }

}
